src/appleseed.studio/mainwindow/renderingsettingswindow.cpp

```diff
@@ -652,6 +662,65 @@ namespace
         }
     };
 
+    class InteractiveImagePlaneSamplingPanel
+      : public ImagePlaneSamplingPanel
+    {
+        Q_OBJECT
+
+      public:
+        InteractiveImagePlaneSamplingPanel(const Configuration& config, QWidget* parent = nullptr)
+          : ImagePlaneSamplingPanel(parent)
+        {
+            QVBoxLayout* layout = new QVBoxLayout();
+            container()->setLayout(layout);
+
+            create_image_plane_sampling_general_settings(layout);
+
+            load_general_sampler(config);
+        }
+
+        void save_config(Configuration& config) const override
+        {            
+            if (get_widget<bool>("general.unlimited_samples"))
+                config.get_parameters().remove_path("progressive_frame_renderer.max_average_samples_per_pixel");
+            else set_config(
+                    config,
+                    "progressive_frame_renderer.max_average_samples_per_pixel",
+                    get_widget<int>("general.max_average_samples_per_pixel"));
+        }
+
+      private:
+        void create_image_plane_sampling_general_settings(QVBoxLayout* parent)
+        {
+            QGroupBox* groupbox = new QGroupBox("General");
+            parent->addWidget(groupbox);
+
+            QVBoxLayout* layout = new QVBoxLayout();
+            groupbox->setLayout(layout);
+
+            QFormLayout* sublayout = create_form_layout();
+            layout->addLayout(sublayout);
+
+            QSpinBox* max_average_samples_per_pixel = create_integer_input("general.max_average_samples_per_pixel", 1, 1000000, 10000);
+            QCheckBox* unlimited_samples = create_checkbox("general.unlimited_samples", "Unlimited");
+            sublayout->addRow("Max Average Samples Per Pixel:", create_horizontal_group(max_average_samples_per_pixel, unlimited_samples));
+            connect(unlimited_samples, SIGNAL(toggled(bool)), max_average_samples_per_pixel, SLOT(setDisabled(bool)));
+        }
+
+        void load_general_sampler(const Configuration& config)
+        {
+            const int DefaultMaxSamples = 100000;
```

Comment:
Needs updating too... Maybe 64 like the default number of SPP in final mode?

Reply:
Ah, yes.